### PR TITLE
Add exception for CPaaS subscription name

### PR DESCRIPTION
**Pull request by briangallagher**
# Issue link
[JIRA](https://issues.redhat.com/browse/MGDAPI-3089)

# What
Validating the RHOAM install failed because the subscription name generated by CPaaS is random. 

# Verification steps
Unit tests sufficient.
Will be validated through CPaaS

### CODE REVIEW

File: pkg/addon/operator_run.go [hunk 1]
```diff
@@ -51,6 +51,16 @@ func InferOperatorRunType(ctx context.Context, client k8sclient.Client, installa
...
 		return runTypesBySubscription[subscription.Name], nil
+	} else {
+		// If subscription is nil, it could be because the CPaaS build has put a non deterministic name on it.
+		// This code may be temporary if CPaaS changes to deterministic Subscription names
+		exists, err := RhoamCPaaSSubscriptionExists(ctx, client, installation.Namespace)
+		if err != nil {
+			return "", err
+		}
+		if exists {
+			return AddonRunType, nil
```
Comment (psturc) [inline pkg/addon/operator_run.go:62]: Is it addon in case of install type that CVP uses? I thought CVP uses OLM

Comment (briangallagher) [reply pkg/addon/operator_run.go:62]: @sergioifg94 because this is CVP and not added through ManagedTenants maybe this should be OLM?

**Review by sergioifg94 — COMMENTED** [review]
I'm not sure if these changes would fix the issue. I think that if that code is added to the [`GetSubscription`](https://github.com/integr8ly/integreatly-operator/blob/master/pkg/addon/operator_run.go#L122) function (instead of the `InferOperatorRunType` one) it would work though. I'd probably add it after the for loop has ended, if it still hasn't found the subscription using the known names then default to the logic you implemented

File: pkg/addon/operator_run.go (hunk 1 above)
Comment (briangallagher) [reply pkg/addon/operator_run.go]: @psturc We won't need to set the type with the new approach. Can you have another quick review of this?

File: pkg/addon/operator_run.go [hunk 2]
```diff
@@ -65,6 +65,37 @@ func InferOperatorRunType(ctx context.Context, client k8sclient.Client, installa
...
+			logrus.Infof("Found subscription %s", sub.Name)
+			labels := sub.GetLabels()
+			// Looking for specific label that should be on the CPaaS generated subscription
+			_, ok := labels["operators.coreos.com/managed-api-service.redhat-rhoam-operator"]
+			if ok {
+				logrus.Info("Found Rhoam operator subscription")
+				return &sub, nil
+			}
+		}
+	}
```
Comment (sergioifg94) [inline pkg/addon/operator_run.go:93]:
```suggestion

	for _, sub := range subs.Items {
		logrus.Infof("Found subscription %s", sub.Name)
		labels := sub.GetLabels()
		// Looking for specific label that should be on the CPaaS generated subscription
		_, ok := labels["operators.coreos.com/managed-api-service.redhat-rhoam-operator"]
		if ok {
			logrus.Info("Found Rhoam operator subscription")
			return &sub, nil
		}
	}

```

**Review by sergioifg94 — APPROVED** [review]
Looks good, just that tiny suggestion but not going to block it because of it
/lgtm
/approve

**Pull request merged**